File: lesson_1/src/fortran_examples/02_variables.f90
PROGRAM main
! Declarations
IMPLICIT NONE   ! In Fortran, by default the type of variables whose names start with I...N is INTEGER,
                ! and REAL otherwise. It is a legacy of the past and it is strongly advised
                ! to put an implicit none statement in your program and in each module. That statement was added in the Fortran 90 standard.
INTEGER :: age
REAL :: height
COMPLEX :: z
CHARACTER(LEN=10) :: name
LOGICAL :: isStudent

    ! Initializations
    age = 25
    height = 1.75
    z = (3.0, 4.0)
    name = 'Alice'
    isStudent = .TRUE.

    ! Printing
    PRINT *, 'Age:', age
    PRINT *, 'Height:', height
    PRINT *, 'The complex number z is:', z
    PRINT *, 'Name:', name
    PRINT *, 'Is Student:', isStudent
END PROGRAM main

File: lesson_1/src/c_examples/02_variables.c
#include <stdio.h>
#include <complex.h>  // For complex numbers
#include <stdbool.h>  // For boolean type

int main() {
    // Declaration and initialization
    int age = 25;
    float height = 1.75;
    double complex z = 3.0 + 4.0*I;  // Complex number: 3 + 4i
    char name[10] = "Alice";         // String of max length 10
    bool isStudent = true;           // Boolean value

    // Printing
    printf("Age: %d\n", age);
    printf("Height: %.2f\n", height);
    printf("The complex number z is: %.2f + %.2fi\n", creal(z), cimag(z));
    printf("Name: %s\n", name);
    printf("Is Student: %s\n", isStudent ? "true" : "false");

    return 0;
}
